cmd/mesh/manifest.go

```diff
@@ -89,6 +93,27 @@ func genManifests(inFilename string, setOverlayYAML string) (name.ManifestMap, e
 	if err != nil {
 		return nil, err
 	}
+	// for installation path contains version number only
+	isp := mergedICPS.InstallPackagePath
+	if version.IsVersionString(isp) {
+		mergedICPS.InstallPackagePath = fmt.Sprintf(helm.InstallationPathTemplate, isp, isp)
+	}
+	if util.IsHTTPURL(mergedICPS.InstallPackagePath) {
```

Comment:
i suggest moving this block to a helper function. it's too low level detail for the reader at this function level.

Reply:
done